tools/eMMC/functions.sh

```diff
@@ -254,140 +278,166 @@ dd_bootloader () {
+  echo_broadcast "Formating boot partition with mkfs.vfat -F 16 ${destination}p1 -n ${boot_label}"
   LC_ALL=C mkfs.vfat -F 16 ${destination}p1 -n ${boot_label}
-  echo "-----------------------------"
+  echo_broadcast "-------------------------------------------------------------------"
   flush_cache
 }
 
 format_root () {
-  message="mkfs.ext4 ${ext4_options} ${destination}p2 -L ${rootfs_label}" ; broadcast
-  echo "-----------------------------"
+  echo_broadcast "-------------------------------------------------------------------"
+  echo_broadcast "Formating rootfs with mkfs.ext4 ${ext4_options} ${destination}p2 -L ${rootfs_label}"
   LC_ALL=C mkfs.ext4 ${ext4_options} ${destination}p2 -L ${rootfs_label}
-  echo "-----------------------------"
+  echo_broadcast "-------------------------------------------------------------------"
   flush_cache
 }
 
 format_single_root () {
-  message="mkfs.ext4 ${ext4_options} ${destination}p1 -L ${boot_label}" ; broadcast
-  echo "-----------------------------"
+  echo_broadcast "-------------------------------------------------------------------"
+  echo_broadcast "Formating rootfs with mkfs.ext4 ${ext4_options} ${destination}p1 -L ${boot_label}"
   LC_ALL=C mkfs.ext4 ${ext4_options} ${destination}p1 -L ${boot_label}
-  echo "-----------------------------"
+  echo_broadcast "-------------------------------------------------------------------"
   flush_cache
 }
 
 copy_boot () {
-  message="Copying: ${source}p1 -> ${destination}p1" ; broadcast
-  mkdir -p /tmp/boot/ || true
-
-  mount ${destination}p1 /tmp/boot/ -o sync
-
-  if [ -f /boot/uboot/MLO ] ; then
+  #FIXME: Something is fishy about this function
+  local tmp_boot_dir="/tmp/boot"
+  echo_broadcast "-------------------------------------------------------------------"
+  echo_broadcast "Copying: ${source}p1 -> ${destination}p1"
+  echo_broadcast "==> Creating temporary boot directory (${tmp_boot_dir})"
+  mkdir -p ${tmp_boot_dir} || true
+  echo_broadcast "==> Mounting ${destination}p1 to ${tmp_boot_dir}"
+  mount ${destination}p1 ${tmp_boot_dir} -o sync
+
+  if [ -f /boot/uboot/MLO ] && [ -f /boot/uboot/u-boot.img ] ; then
+    echo_broadcast "==> Found MLO and u-boot.img in current /boot/uboot/, copying"
     #Make sure the BootLoader gets copied first:
-    cp -v /boot/uboot/MLO /tmp/boot/MLO || write_failure
+    cp -v /boot/uboot/MLO ${tmp_boot_dir}/MLO || write_failure
     flush_cache
 
-    cp -v /boot/uboot/u-boot.img /tmp/boot/u-boot.img || write_failure
+    cp -v /boot/uboot/u-boot.img ${tmp_boot_dir}/u-boot.img || write_failure
     flush_cache
+  else
+    echo_broadcast "!==> We could not find an MLO and u-boot.img to copy"
   fi
 
-  message="rsync: /boot/uboot/ -> /tmp/boot/" ; broadcast
-  rsync -aAx /boot/uboot/ /tmp/boot/ --exclude={MLO,u-boot.img,uEnv.txt} || write_failure
+  echo_broadcast "==> rsync: /boot/uboot/ -> ${tmp_boot_dir}"
+  #FIXME: Why is it rsyncing /boot/uboot ? This is wrong
```

Comment:
Humm, OK 
I'm reworking all this currently because I need a 2 layout partition for implementing encryption of the rootfs.
I'm therefor bringing back the /boot capability